diff --git a/CMakeLists.txt b/CMakeLists.txt
index aec7643..2b32092 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -21,26 +21,8 @@ else()
   set(PATCHELF_BUILD_COMMAND "${MAKE_EXE}")
 endif()
 
-ExternalProject_add(build_patchelf
-  SOURCE_DIR ${Patchelf_SOURCE_DIR}
-  DOWNLOAD_COMMAND ""
-  UPDATE_COMMAND ""
-  PATCH_COMMAND ./bootstrap.sh
-  CONFIGURE_COMMAND ./configure
-  BUILD_COMMAND ${PATCHELF_BUILD_COMMAND}
-  BUILD_IN_SOURCE 1
-  INSTALL_COMMAND ""
-)
-set(patchelf_executable ${Patchelf_SOURCE_DIR}/src/patchelf${CMAKE_EXECUTABLE_SUFFIX})
-
-find_program(STRIP_EXECUTABLE strip)
-if(STRIP_EXECUTABLE)
-  add_custom_target(strip_patchelf_executable ALL
-    COMMAND ${STRIP_EXECUTABLE} ${patchelf_executable}
-    WORKING_DIRECTORY ${Patchelf_SOURCE_DIR}
-    COMMENT "Stripping patchelf executable"
-    )
-  add_dependencies(strip_patchelf_executable build_patchelf)
-endif()
+find_program(SYSTEM_PATCHELF_BIN NAMES patchelf REQUIRED)
+message(STATUS "Found system patchelf: ${SYSTEM_PATCHELF_BIN}")
+set(patchelf_executable ${SYSTEM_PATCHELF_BIN})
 
 install(PROGRAMS "${patchelf_executable}" COMPONENT python DESTINATION "${SKBUILD_SCRIPTS_DIR}")
